### ISSUE TITLE
Translated entries data lost when adding a new field

### ISSUE THREAD

**Issue by sebszocinski** [statamic/v2-hub#1987]
**The issue:**
When adding a new field to an existing fieldset (in my case in a collection) where entries have already been previously translated in the past. Some (not all) translated entries data is wiped when saving the default language via the CP (once new filed is added).

**Steps to reproduce:**
1. Have existing collection with translated entries in multiple fields.
2. Create a new field via the yaml files (translatable)
3. Edit the default field in either the CP or via the index.md file but hit save in the CP
4. Watch some of your existing translated md files in your editor go yellow (changed) and when opening one you'll see that ALL data is lost except for the ID.

**Expected behavior**
Data is not lost! major issue! if this happens on the production server and my client loses all their translations they will be pissed!

**Screenshots**

**Environment details (please complete the following information):**
 - This is happening on Statamic 2.8.12 and as far as I can see no updates or fixes to translations have been made in recent updates?
 - Fresh Install or Upgrade:
 - OS: MacOS
 - Browser: Chrome
 - Web Server: Valet
 - PHP Version: 7.1
 - Addons installed: None

**Additional context**

**Comment by jasonvarga** [comment]
Is this any different to #1222 ?

**Comment by sebszocinski** [comment]
Yeah I guess it is but that issue is quite old now so I thought i'd start a new one... any solutions to this one yet?

**Comment by jasonvarga** [comment]
I can't reproduce it. Can you give more details? A video of it?

**Comment by sebszocinski** [comment]
Yep - here you go: https://cl.ly/3X3S262Z3g2r

**Comment by jasonvarga** [comment]
What is your default locale?
Can you send me the fieldset?

**Comment by sebszocinski** [comment]
```
locales:
  au:
    name: Australia
    full: en_AU
    url: "{env:AU_URL}"
```

and in `env`

`AU_URL="http://sitename.test"`

**Comment by sebszocinski** [comment]
It's not limited to a single fieldset either... i've seen it happen on multiple pages / collections that use different fieldsets.

**Comment by jasonvarga** [comment]
If you reduce the fieldset down to only a few text fields and try it again, does it still happen?

**Comment by sebszocinski** [comment]
Yep, I just removed all fields except for the content (redactor) and a simple text field and tested it and only a few translated versions were wiped (not all)... tested the exact same thing again and this time around a DIFFERENT bunch of translated versions were wiped:

First test:
<img width="253" alt="screen shot 2018-06-07 at 9 27 24 am" src="https://user-images.githubusercontent.com/5466694/41071089-62faf582-6a35-11e8-8e2f-b665525a2c76.png">

Second test:
<img width="223" alt="screen shot 2018-06-07 at 9 28 36 am" src="https://user-images.githubusercontent.com/5466694/41071091-669e9c7a-6a35-11e8-999d-5bc81f34f36d.png">

Very bizarre but also worrying... I'm happy to share a copy of the site or give you temp access to the repo if you like?
